frontend/src/pages/modelServing/screens/projects/kServeModal/ManageKServeModal.tsx

```diff
@@ -139,6 +139,11 @@ const ManageKServeModal: React.FC<ManageKServeModalProps> = ({
     namespace,
   });
 
+  const [focusTarget, setFocusTarget] = React.useState<string | null>(null);
```

Comment:
I think having `focusTarget` as React state may be unnecessary complexity.

Can we try this:
* Lift your `inputRef` out to this component as `servingRuntimeArgsInputRef`, pass that down as a prop into `servingRuntimeArgsSection` to attach it to the input. No further logic necessary in that component.
* Instead of passing an `onFocusChange` into `ServingRuntimeTemplateSection`, pass an `onConfigureParamsClick` callback that directly calls your `requestAnimationFrame(() => { inputRef.current?.focus() })`

Then we don't need any new state.